### PR TITLE
depend on docker-selinux for fedora-25+ and centos-7

### PR DESCRIPTION
**Pull request by andrewhsu**
**- What I did**

Changed the dependency in the docker-engine RPM built for Fedora 25+ to depend on `docker-selinux` which is satisfied by the `container-selinux` package. This PR fixes issue #28099.

Also, now that [centos 7.3 has been released](https://lists.centos.org/pipermail/centos-announce/2016-December/022172.html), changed the dependency in the `docker-engine` RPM built for CentOS to depend on `docker-selinux` which happens to be `docker-selinux-1.10.3-46.el7.centos.14.x86_64` but should still work for docker-engine all the way up through 1.13.

**- How I did it**

Added a conditional `Requires` in the `docker-engine.spec` file for fedora-25.

Added a section to the `docker-engine.spec` file for defining a subpackage. This will cause centos-7 builds to create a `docker-engine-selinux` [subpackage](http://rpm.org/max-rpm-snapshot/ch-rpm-subpack.html) (which is just another RPM built from the same spec file) that has no files but just a dependency on the `docker-selinux` package. 

Also refactored the `hack/make/build-rpm` to only build a `docker-engine-selinux` package if there exists a specific selinux policy directory in the `contrib` dir. In doing so, found that the `contrib/selinux` policy dir is no longer used by anybody so deleted it.

Note: this changes the default behaviour of building RPMs from using a generic selinux policy to only using one for the specificy distro version.

**- How to verify it**

Build the fedora-25 rpm and install on a Fedora 25 machine with selinux enabled.

```
$ make DOCKER_BUILD_PKGS=fedora-25 rpm
```

Build a `docker-engine` centos-7 RPM package and install on centos 7.3 with selinux enabled:
```
$ make DOCKER_BUILD_PKGS=centos-7 rpm
```

Configure docker daemon to run with selinux by setting `/etc/docker/daemon.json`:
```json
{
  "selinux-enabled": true
}
```

See that the `svirt_sandbox_file_t` label is set which means docker is doing the right thing with selinux enabled:
```
$ sudo docker run --rm ubuntu ls -alZ /home
total 4
drwxr-xr-x.  2 root root system_u:object_r:svirt_sandbox_file_t:s0:c407,c628    6 Apr 12  2016 .
drwxr-xr-x. 21 root root system_u:object_r:svirt_sandbox_file_t:s0:c407,c628 4096 Dec 14 04:49 ..
```

**- Description for the changelog**

Use the selinux policy provided by `docker-selinux` on Fedora 25+ and CentOS 7.3.

**- A picture of a cute animal (not mandatory but encouraged)**

🐺

### CODE REVIEW

**Comment by andrewhsu** [comment]
Right now, building rpms for fedora 25 is not working so this PR will be difficult to verify. I have a fix: #29115.

**Comment by andrewhsu** [comment]
/cc @crosbymichael @tiborvass 

I was able to verify this by building a fedora 25 rpm with the help of #29115. Installing the rpm pulls in `container-selinux` with the selinux policies for docker:

```
$ sudo yum -y install docker-engine-*.rpm
...
Installed:
  container-selinux.x86_64 2:1.12.3-10.git7b5044b.fc25          docker-engine.x86_64 1.14.0-0.0.20161205.024700.git8b634c0.fc25
  iptables.x86_64 1.6.0-2.fc25                                  libnetfilter_conntrack.x86_64 1.0.4-6.fc24
  libnfnetlink.x86_64 1.0.1-8.fc24                              libtool-ltdl.x86_64 2.4.6-12.fc25
  policycoreutils-python-utils.x86_64 2.5-18.fc25

Complete!
```

**Comment by cpuguy83** [comment]
Cool, so if I read this right, instead of generating our own selinux package we'll use redhat's for anything that doesn't require some specific change?

File: hack/make/build-rpm [hunk 1]
```diff
@@ -121,13 +121,12 @@ set -e
...
-		# selinux policy referencing systemd things won't work on non-systemd versions
-		# of centos or rhel, which we don't support anyways
-		if [ "${suite%.*}" -gt 6 ] && [[ "$version" != opensuse* ]]; then
-			selinuxDir="selinux"
-			if [ -d "./contrib/selinux-$version" ]; then
-				selinuxDir="selinux-${version}"
-			fi
+
+		# by default do not create selinux rpm
+		# if there exists a selinux policy dir specifically for distro verison
```
Comment (thaJeztah) [inline hack/make/build-rpm]: typo: `verison` -> `version`

Comment (andrewhsu) [inline hack/make/build-rpm]: werd

**Comment by andrewhsu** [comment]
fyi increased the scope of this PR from fedora-25 to also include centos-7

**Comment by cpuguy83** [comment]
This is not working at least on centos7, haven't tested fedora yet.

**Comment by andrewhsu** [comment]
@cpuguy83 wondering if the `vendor.conf` needs to be updated to a newer git commit hash of `opencontainers/runc` to bring in this commit related to selinux: https://github.com/opencontainers/runc/commit/a4f3123c355962945f5a0b174f7b4f9804bf3fc3

**Comment by thaJeztah** [comment]
ping @mlaventure ^^

**Comment by cpuguy83** [comment]
@andrewhsu I don't think this is going to fix our issue here.

**Comment by mlaventure** [comment]
It worked for me on a centos7.3 DO Droplet

**Comment by cpuguy83** [comment]
@mlaventure DO doesn't have selinux enabled. Check `docker info` under security options, no selinux. This is because runc doesn't see any selinuxfs mounted.

**Comment by LK4D4** [comment]
ping @andrewhsu @cpuguy83 @mlaventure

**Comment by runcom** [comment]
ping @andrewhsu @cpuguy83 any resolution with this? anything I can help you with?

**Comment by andrewhsu** [comment]
I'm going to close this for now since it does not work on all versions of centos7. This was my attempt to make building selinux packages a bit easier but it is not crucial for the proper functioning of selinux with docker.

**Pull request closed without merging**